fuel-tx/src/transaction/types/blob.rs

```diff
@@ -0,0 +1,250 @@
+    }
+}
+
+impl UniqueFormatValidityChecks for Blob {
+    fn check_unique_rules(
+        &self,
+        consensus_params: &ConsensusParameters,
+    ) -> Result<(), ValidityError> {
+        let index = self.body.witness_index as usize;
+        let witness = self
+            .witnesses
+            .get(index)
+            .ok_or(ValidityError::InputWitnessIndexBounds { index })?;
+
+        // Verify that blob id is correct
+        if BlobId::compute(witness.as_ref()) != self.body.id {
+            return Err(ValidityError::TransactionBlobIdVerificationFailed);
+        }
+
+        self.inputs
+            .iter()
+            .enumerate()
+            .try_for_each(|(index, input)| {
+                if let Some(asset_id) = input.asset_id(consensus_params.base_asset_id()) {
+                    if asset_id != consensus_params.base_asset_id() {
+                        return Err(
+                            ValidityError::TransactionInputContainsNonBaseAssetId {
+                                index,
+                            },
+                        );
+                    }
+                }
+
+                match input {
+                    Input::Contract(_) => {
+                        Err(ValidityError::TransactionInputContainsContract { index })
+                    }
+                    Input::MessageDataSigned(_) | Input::MessageDataPredicate(_) => {
+                        Err(ValidityError::TransactionInputContainsMessageData { index })
+                    }
+                    _ => Ok(()),
+                }
+            })?;
+
+        self.outputs
+            .iter()
+            .enumerate()
+            .try_for_each(|(index, output)| match output {
+                Output::Contract(_) => {
+                    Err(ValidityError::TransactionOutputContainsContract { index })
+                }
+
+                Output::Variable { .. } => {
+                    Err(ValidityError::TransactionOutputContainsVariable { index })
+                }
+
+                Output::Change { asset_id, .. } => {
+                    if asset_id != consensus_params.base_asset_id() {
+                        Err(ValidityError::TransactionChangeChangeUsesNotBaseAsset {
+                            index,
+                        })
+                    } else {
+                        Ok(())
+                    }
+                }
+
+                Output::ContractCreated { .. } => {
+                    Err(ValidityError::TransactionOutputContainsContractCreated { index })
+                }
+                Output::Coin { asset_id, .. } => {
+                    if asset_id != consensus_params.base_asset_id() {
+                        Err(ValidityError::TransactionInputContainsNonBaseAssetId {
+                            index,
+                        })
+                    } else {
+                        Ok(())
+                    }
+                }
```

Comment:
We don't need this branch because combination of the `TransactionOutputCoinAssetIdNotFound` and `TransactionInputContainsNonBaseAssetId` guarantees it=)